apps/gitness/src/pages/project-settings/project-settings-new-member-page.tsx

```diff
@@ -0,0 +1,214 @@
+                    {usersData?.length !== 0 ? (
+                      'Available users'
+                    ) : (
+                      <Text color="tertiaryBackground">No available users</Text>
+                    )}
+                  </DropdownMenuLabel>
+                  {usersData && <DropdownMenuSeparator />}
+                  {usersData &&
+                    usersData.map(user => (
+                      <DropdownMenuItem
+                        key={user.uid}
+                        onSelect={() => handleMemberSelect(user.uid ?? '', user.display_name ?? '')}>
+                        {user.display_name}
+                      </DropdownMenuItem>
+                    ))}
+                </DropdownMenuContent>
+              </DropdownMenu>
+              {/* Register the field for validation */}
+              <input type="hidden" {...register('memberName', { required: 'Please select a member' })} />
+              {errors.memberName && (
+                <FormFieldSet.Message theme={FormFieldSet.MessageTheme.ERROR}>
+                  {errors.memberName.message}
+                </FormFieldSet.Message>
+              )}
+            </FormFieldSet.ControlGroup>
+
+            {/* Role Selection */}
+            <FormFieldSet.ControlGroup>
+              <FormFieldSet.Label htmlFor="role" required>
+                Role
+              </FormFieldSet.Label>
+              <Select
+                value={newMemberRoleValue}
+                onValueChange={value => setValue('role', value, { shouldValidate: true })}
+                disabled={isSubmitting}>
+                <SelectTrigger id="role">
+                  <SelectValue placeholder="Select" />
+                </SelectTrigger>
+                <SelectContent>
+                  {roleOptions.map(role => (
+                    <SelectItem key={role.value} value={role.value}>
+                      {role.label}
+                    </SelectItem>
+                  ))}
+                </SelectContent>
+              </Select>
+              {errors.role && (
+                <FormFieldSet.Message theme={FormFieldSet.MessageTheme.ERROR}>
+                  {errors.role.message}
+                </FormFieldSet.Message>
+              )}
+            </FormFieldSet.ControlGroup>
+
+            {/* Action Buttons */}
+            <FormFieldSet.ControlGroup>
+              <ButtonGroup.Root>
+                {!submitted ? (
+                  <>
+                    <Button size="sm" type="submit" disabled={!isValid || isSubmitting}>
+                      {isSubmitting ? 'Inviting...' : 'Invite New Member'}
+                    </Button>
+                    <Button size="sm" variant="outline" onClick={handleCancel} disabled={isSubmitting}>
+                      Cancel
+                    </Button>
+                  </>
+                ) : (
+                  <Button variant="ghost" size="sm" theme="success" className="pointer-events-none">
```

Comment:
we dont need this state. let use change the form and click on "save" again.

Reply:
deleted